src/authenticator.ts

```diff
@@ -93,18 +98,23 @@ export default class Authenticator {
     });
   }
 
-  private generateAccessToken(options: AuthenticateOptions): string {
+  generateAccessToken(options: AuthenticateOptions): TokenWithExpire {
```

Comment:
I see that this would leave us with a token that is either an object (Access Token / TokenWithExpire), or a string (RefreshToken). 

I think we should aim for more consistency - make refresh token an object as well perhaps?